tests/test_main.py

```diff
@@ -729,6 +729,10 @@ def test_sample_files(self, full_filename, result_filename):
                 assert result == json.load(infile)
 
 
+    @pytest.mark.skipif(
+        NodeFinder.__name__ == "SentinelNodeFinder",
+        reason="The SentinelNodeFinder has problems in some situations (see skip_sentinel)",
```

Comment:
This is a huge amount of testing to skip for everything <3.11

Reply:
I think the problem appeared when I added the ipython and coverage dependencies.
Some files trigger the same issues which gets triggered by `test_small_samples`.
This small_samples are already excluded for the SentinelNodeFinder (`skip_sentinel` array).

I don't know how easy it is to fix this issues in the SentinelNodeFinder.
I don't want to put to much time into this, because I don't have to much and because #64 is already pretty far and can replace the SentinelNodeFinder for 3.8, 3.9 and 3.10.

I also created https://github.com/15r10nk/pysource-codegen with the coal to use it to test executing. I'm pretty sure that I will find more issues in all of the 3 node finders. The question here is if it is possible to solve them all.

It will be some work, but based on my experience with Index/PositionNodefinder, I'm pretty sure they are solvable.
However I have no experience with the SentinelNodeFinder. What do you think?

Reply:
I'm currently working on the SentinelNodeFinder and try to fix some Issues to get a better understanding